### PR TITLE
Added script to cleanup resources

### PR DESCRIPTION
**Pull request by ivanmkc**
Added cleanup script for Vertex AI resources

### CODE REVIEW

File: .cloud-build/cleanup/resource_cleanup_manager.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        pass
+
+    def is_deletable(self, resource: Any) -> bool:
+        time_difference = self.get_seconds_since_modification(resource)
+
+        if self.resource_name(resource).startswith("perm_"):
+            print(f"Skipping '{resource}' due to name starting with 'perm_'.")
+            return False
+
+        # Check that it wasn't created too recently, to prevent race conditions
```
Comment (ivanmkc) [inline .cloud-build/cleanup/resource_cleanup_manager.py:40]: Prevent deletions of resources currently in use by a test

Comment (ivanmkc) [reply .cloud-build/cleanup/resource_cleanup_manager.py:40]: I think the service should prevent this though

File: .cloud-build/cleanup/resource_cleanup_manager.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+    @abc.abstractmethod
+    def get_seconds_since_modification(self, resource: Any) -> float:
+        pass
+
+    def is_deletable(self, resource: Any) -> bool:
+        time_difference = self.get_seconds_since_modification(resource)
+
+        if self.resource_name(resource).startswith("permanent"):
+            print(f"Skipping '{resource}' due to name starting with 'perm_'.")
```
Comment (davidcavazos) [inline .cloud-build/cleanup/resource_cleanup_manager.py:37]: `perm_` --> `permanent`

**Review by leahecole — APPROVED** [review]

**Pull request merged**